tests/e2e-prometheuscr/create-sm-prometheus-exporters/01-assert.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/instance: create-sm-prometheus.simplest
+    app.kubernetes.io/managed-by: opentelemetry-operator
+    app.kubernetes.io/name: prometheus-dev-collector
+  name: prometheus-dev-collector
+  namespace: create-sm-prometheus
+spec:
```

Comment:
The prometheus adds additional labels during scraping e.g. namespace, pod, container, instance, endpoint see https://github.com/openshift/openshift-docs/pull/60953/files#diff-bd0dc510dd646c60e63a7daf0b9d22eb7525adc8467753a6bdb37948345992ddR9

these labels will identify the OTEL collector. 

Similar labels k8s_{namespace, pod} etc. will be collected as well by OTEL SDKS (or collector) shall we offer the option to remove the prometheus labels?

Reply:
FYI, if you use the prometheus receiver instead, you'll get these labels (though using the Otel semantic convention) even if you delete them in the ServiceMonitor. This is because prometheus receiver uses the parsing context directly, see: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/internal/prom_to_otlp.go.

The only way to get rid of them is to remove them in a processor after the fact. I don't think this is a problem in this case, as they should be identical to the attributes from instrumentation. Maybe we should make this consistent and just rename the Prometheus-provided attributes to the Otel schema here?

Reply:
I tried to rename the labels to the OTEL Schema but... the original labels keep appearing.
Should I remove labels as per the link provided by @pavolloffay? Even when they will be "recovered" if using the Prometheus receiver. Or should we ignore them?

Reply:
It would be very strange if you could remove, but not rename them. In the interest of not holding up this PR, the former is ok with me.

Reply:
@swiatekm-sumo maybe I'm not using the correct rules or something. I tried with `labelmap` and `replace` actions. Both of them created a new label but not removed the old one.
I was not able to find a better solution.

Reply:
Looking at the changes, you did the following:

```yaml
- action: replace
  regex: container
  replacement: k8s.container.name
```

but what this action does is match the regex against label *values*, and then copies the *value* of that label to the target. You need to drop the original separately. Try this:

```yaml
- source_labels: [pod]
  action: replace
  target_label: k8s.pod.name
  regex: (.+)
- action: labeldrop
  regex: ^pod$
```

Reply:
I was talking to @pavolloffay offline and I'll create a new task to address this as part of a separate issue because we're not sure if we want to suggest removing or relabeling.
Thanks for your comment (I'm curious and I'll try it to learn)